src/mixins/object_geometry.mixin.js

```diff
@@ -601,77 +601,51 @@
       return cache.value;
     },
 
-    /*
+    /** 
      * Calculate object dimensions from its properties
      * @private
-     * @return {Object} .x width dimension
-     * @return {Object} .y height dimension
+     * @returns {fabric.Point} dimensions
      */
     _getNonTransformedDimensions: function() {
-      var strokeWidth = this.strokeWidth,
-          w = this.width + strokeWidth,
-          h = this.height + strokeWidth;
-      return { x: w, y: h };
+      return new fabric.Point(this.width, this.height).scalarAddEquals(this.strokeWidth);
     },
 
-    /*
+    /**
      * Calculate object bounding box dimensions from its properties scale, skew.
-     * @param {Number} skewX, a value to override current skewX
-     * @param {Number} skewY, a value to override current skewY
+     * @param {Object} [options]
+     * @param {Number} options.scaleX a value to override current skewX
+     * @param {Number} options.scaleY a value to override current skewX
+     * @param {Number} options.skewX a value to override current skewX
+     * @param {Number} options.skewY a value to override current skewY
      * @private
-     * @return {Object} .x width dimension
-     * @return {Object} .y height dimension
+     * @returns {fabric.Point} dimensions
      */
-    _getTransformedDimensions: function(skewX, skewY) {
-      if (typeof skewX === 'undefined') {
-        skewX = this.skewX;
-      }
-      if (typeof skewY === 'undefined') {
-        skewY = this.skewY;
-      }
-      var dimensions, dimX, dimY,
-          noSkew = skewX === 0 && skewY === 0;
+    _getTransformedDimensions: function (options) {
+      options = Object.assign({
+        scaleX: this.scaleX,
+        scaleY: this.scaleY,
+        skewX: this.skewX,
+        skewY: this.skewY,
+      }, options || {});
+      var dimX = this.width, dimY = this.height, finalDimensions,
+        noSkew = options.skewX === 0 && options.skewY === 0;
 
-      if (this.strokeUniform) {
-        dimX = this.width;
-        dimY = this.height;
+      if (noSkew) {
+        finalDimensions = new fabric.Point(dimX * options.scaleX, dimY * options.scaleY);
       }
       else {
-        dimensions = this._getNonTransformedDimensions();
```

Comment:
@asturur was this call intentional?
Seems wrong to me because in some cases `strokeWidth` will be accounted for twice.
This could be what was causing some of the problems with `strokeUniform`

Reply:
my mistake